Chimera master bc4 feature 6534

rommstein opened the pull request.

rommstein committed:
BJS2-7339
post_service

rommstein committed:
BJS2-7445
post_service
вынес реализацию commentEvent в маппер

rommstein committed:
BJS2-6534

rommstein committed:
BJS2-6534
посты + комменты + лайки./

rommstein committed:
BJS2-6534
посты + комменты + лайки
Refactoring

InnMish commented on src/main/java/faang/school/postservice/config/threadpool/ThreadPoolConfig.java:
```diff
@@ -0,0 +1,20 @@
+
+    @Bean
+    public ExecutorService threadPool() {
+        return Executors.newFixedThreadPool(threadCount);
```
Spring предоставляет интерфейс TaskExecutor, который является абстракцией над ThreadPoolTaskExecutor.  ThreadPoolTaskExecutor является реализацией этого интерфейса, и спринг умеет им управлять сам
если мы используем потоки через Executors, мы обязаны руками управлять потоками
если же через ThreadPoolTaskExecutor - управлять будет спринг

InnMish commented on src/main/java/faang/school/postservice/controller/CommentController.java:
```diff
@@ -22,30 +21,26 @@
-    public CommentDto createComment(@PathVariable long postId, @Valid @RequestBody CommentDto commentDto){
-        commentDto.setPostId(postId);
-        return commentService.createComment(commentDto);
+    @PostMapping("/create/{postId}")
```
HTTP-метод POST явно указывает, что ты создаешь новый комментарий
В REST API URL-ы должны представлять собой ресурсы. В данном случае комментарий является ресурсом, и он относится к посту 
то есть здесь мы сущность описываем, а не действие 

@PostMapping("/post/{postId}/comment") так например

InnMish commented on src/main/java/faang/school/postservice/controller/CommentController.java:
```diff
@@ -22,30 +21,26 @@
-    @PutMapping("/{commentId}")
-    @ResponseStatus(HttpStatus.OK)
-    public CommentDto updateComment(@PathVariable long commentId,@Valid @RequestBody CommentDto commentDto){
+    @PutMapping("/update/{commentId}")
```
глаголы в урлах не пишем, такой же коммент, как выше

InnMish commented on src/main/java/faang/school/postservice/controller/PostController.java:
```diff
@@ -0,0 +1,71 @@
+
+    private final PostService postService;
+
+    @PostMapping("/create")
```
такой же коммент, как в контроллере комментов

InnMish commented on src/main/java/faang/school/postservice/mapper/PostMapper.java:
```diff
@@ -3,10 +3,18 @@
+    @Mapping(target = "ad", ignore = true)
+    @Mapping(target = "resources", ignore = true)
+    @Mapping(target = "verified", ignore = true)
+    @Mapping(target = "verifiedAt", ignore = true)
```
чтобы все поля не игнорить на 8 строчки допиши property = таргет полиси стретеджи игнор 
что-то такое там есть

InnMish commented on src/main/java/faang/school/postservice/service/comment/CommentServiceImpl.java:
```diff
@@ -0,0 +1,81 @@
+        return postRepository.existsById(id);
+    }
+
+    private Comment getComment(Long userId, Long postId, CommentDto commentDto) {
```
название метода не говорящее 
get - это получить откуда-то, а мы делаем create

InnMish commented on src/main/java/faang/school/postservice/service/comment/CommentServiceImpl.java:
```diff
@@ -0,0 +1,81 @@
+        return comment;
+    }
+
+    private Post getPost(Long postId) {
```
вот это в пост сервисе метод же должен быть, а не тут 
из сервиса постов просто дерни нужный тебе метод (этот), чтобы не дублировать код

InnMish commented on src/main/java/faang/school/postservice/service/post/PostServiceImpl.java:
```diff
@@ -57,44 +66,100 @@ private List<List<Post>> createSubLists(List<Post> posts) {
+                        post.setPublishedAt(LocalDateTime.now());
+                    });
+                }, threadPool));
+    }
```
```
List<Post> posts = postRepository.findReadyToPublish();

    // Делим список постов на батчи и публикуем их асинхронно
    for (int i = 0; i < posts.size(); i += scheduledPostsBatchSize) {
        List<Post> batch = posts.subList(i, Math.min(i + scheduledPostsBatchSize, posts.size()));
        CompletableFuture.runAsync(() -> {
            batch.forEach(post -> {
                post.setPublished(true);
                post.setPublishedAt(LocalDateTime.now());
                postRepository.save(post);
            });
        }, threadPool);
    }
```
    
    Ну будет проще? 
    
    2. тебе тут не нужен threadPool бином 
    пропиши над методом аннотацию Async и положи в нее название бина threadPool, который ты в конфиге собрал

rommstein committed:
BJS2-6534
Поправил аннотации чтобы было всё по REST.
Исправил ThreadPoolConfig.
Добавил аннотацию в Маппер
Поменял названеи методов
Изменил publishScheduledPosts

rommstein committed:
Merge branch 'chimera-master-bc4' into chimera-master-bc4-feature-6534
